consul/base.py

```diff
@@ -1831,6 +1852,8 @@ def list(self, index=None, wait=None, consistency=None, dc=None):
             dc = dc or self.agent.dc
             if dc:
                 params.append(('dc', dc))
+            if token:
```

Comment:
```suggestion
            token = token or self.agent.token
            if token:
```
agent may already have token set. Use it!